### PR TITLE
Irregular binning option in likelihood

### PR DESCRIPTION
**Pull request by xzh19980906**
This PR adds an option bins_type = 'irreg', in which bins should be a tuple with length 2. Same as the return of `appletree.utils.get_equiprob_bins_2d`, the first element in the tuple should be a 1D array that defines the bin edges of the 1st axis, and the second element should be a 2D array that defines the bin edges of the 2nd axis in each slice.

After this PR, in the likelihood config, you can do
```
likelihood_config = {
    ...
    "bins_type": "irreg",
    "bins_on": ["xxx", "yyy"],
    "bins": ([1,2,3],
             [[0, 30, 60, 100], [0, 30, 60, 100]]),
    "x_clip": [1, 3],
    "y_clip": [0, 100],
}
```

This could be useful when you don't like the binning by the data-equiprobable-binning algorithm, for example when binning the integers.

### CODE REVIEW

File: appletree/likelihood.py [hunk 1]
```diff
@@ -71,6 +71,16 @@ def __init__(self, name: str = None, **config):
...
+                raise RuntimeError('only 2D irregular binned likelihood is supported!')
+            self.component_bins_type = 'irreg'
+            self.data_hist = make_hist_irreg_bin_2d(
+                self.data,
+                bins_x=jnp.array(self._bins[0]),
+                bins_y=jnp.array(self._bins[1]),
+                weights=jnp.ones(len(self.data)),
+            )
         else:
             raise ValueError("'bins_type' should either be meshgrid or equiprob")
```
Comment (dachengx) [inline appletree/likelihood.py:85]: Here we need to change the message a bit.

**Comment by dachengx** [comment]
Looks good. I will add several lines to check the binning.

**Comment by dachengx** [comment]
In the further we can also offer the option that first applies binning in the y-axis, then the x-axis.

**Comment by dachengx** [comment]
Also, `bins[0]` and `bins[1]` should not have the same length, according to https://github.com/XENONnT/appletree/blob/irreg_bin/appletree/hist.py#L23. I have edited the description of this PR.

**Comment by dachengx** [comment]
@xzh19980906 Looks good to go.

**Review by dachengx — APPROVED** [review]
